neuro_san/internals/utils/resolver_util.py

```diff
@@ -0,0 +1,74 @@
+# Copyright (C) 2023-2025 Cognizant Digital Business, Evolutionary AI.
+# All Rights Reserved.
+# Issued under the Academic Public License.
+#
+# You can be released from the terms, and requirements of the Academic Public
+# License by purchasing a commercial license.
+# Purchase of a commercial license is mandatory for any use of the
+# neuro-san SDK Software in commercial settings.
+#
+# END COPYRIGHT
+
+from typing import Any
+from typing import List
+from typing import Type
+
+from leaf_common.config.resolver import Resolver
+
+
+class ResolverUtil:
+    """
+    Helper class for creating instances of classes that are referenced by a string.
+    """
+
+    @staticmethod
+    def create_instance(class_name: str, class_name_source: str, type_of_class: Type) -> Any:
+        """
+        Resolves the class_name to instantiate an instance of the class.
+        Goes through some standard checking with standard exceptions thrown
+        to make sure the class is set up the way we want it.
+
+        :param class_name: The fully qualified class name to instantiate
+        :param class_name_source: A string description of where we are getting the value of
+                    class_name, so that exceptions can be more instructive.
+        :param type_of_class: The type that the instance must be in order to pass muster.
+        :return: An instance of the class referred to by class_name if everything is successful.
+                Can return None if class_name is a None or empty string.
+        """
```

Comment:
This helper method contains an abstraction of commonly repeated code to instantiate a class of a particular type.
Eventually we might use this in other places (thinking DefaultLlmFactory here, from where this was largely lifted), but making such changes is out of scope for this PR.